### PR TITLE
Document ability to add custom filesystems

### PR DESCRIPTION
**Pull request by brenelz**
@GrahamCampbell @mattstauffer @barryvdh Sorry about the pull request spam guys.  Just learning a bit about squashing commits and was having trouble.  I think I got it figured out now, so I'll keep this one open and make updates to it as required.

### CODE REVIEW

**Comment by brenelz** [comment]
Sorry about the pull request spam guys.  Just learning a bit about squashing commits and was having trouble.  I think I got it figured out now, so I'll keep this one open and make updates to it as required.

File: filesystem.md [hunk 1]
```diff
@@ -106,3 +107,41 @@ The `Storage` facade may be used to interact with any of your configured disks.
 #### Delete A Directory
 
 	Storage::deleteDirectory($directory);
+
+<a name="custom-filesystems"></a>
+## Custom Filesystems
+Laravel's Flysystem integration provides drivers for local filesystems, Amazon S3, and Rackspace Cloud Storage. Flysystem is not limited to these and has adapters for many additional storage systems. If you want to use one of these additional adapters in your Laravel application, you'll want to create a custom filesystem.
```
Comment (barryvdh) [inline filesystem.md]: > you'll want to create a custom filesystem.
You want to create a custom driver, not filesystem.

File: filesystem.md [hunk 2]
```diff
@@ -106,3 +107,41 @@ The `Storage` facade may be used to interact with any of your configured disks.
...
+
+In order to set up your custom filesystem you'll need to create a Service Provider (for example, `DropboxFilesystemServiceProvider`). In its `boot` method, you'll want to inject an instance of the `Illuminate\Contracts\Filesystem\Factory` contract, and then call the `extend` method of the injected instance.
+
+The first argument of the `extend` method is the name of the disk you will use, while the second argument is a closure that gets passed both the `$app` and `$config` variables.
```
Comment (barryvdh) [inline filesystem.md]: First argument is the name of the driver.

**Comment by barryvdh** [comment]
Do you also need to add a note that, in order to actually use the filesystem, you need to use the `dropbox` driver in the config, instead of local/s3? Or is that obvious?

**Comment by brenelz** [comment]
You can either change the default driver, or you can use the disk method.  It isn't different from using other drivers but we could add a note.

**Comment by mattstauffer** [comment]
Nice work!

**Pull request merged**